src/backend/distributed/executor/multi_server_executor.c

```diff
@@ -98,13 +99,26 @@ JobExecutorType(DistributedPlan *distributedPlan)
 									  "\"task-tracker\".")));
 		}
 
-		/* if we have repartition jobs with real time executor, error out */
+		/* if we have repartition jobs with real time executor and repartition
+		 * joins are not enabled, error out. Otherwise, switch to task-tracker
+		 */
 		dependedJobCount = list_length(job->dependedJobList);
 		if (dependedJobCount > 0)
 		{
-			ereport(ERROR, (errmsg("cannot use real time executor with repartition jobs"),
-							errhint("Set citus.task_executor_type to "
-									"\"task-tracker\".")));
+			if (!EnableRepartitionJoin)
+			{
+				ereport(ERROR, (errmsg(
+									"cannot use real time executor with repartition jobs"),
+								errhint("Set citus.task_executor_type to "
+										"\"task-tracker\" or "
+										"set citus.enable_repartition_joins to on.")));
```

Comment:
Looking at this error message again, it contains a lot of terminology, maybe we could simplify it to:

```
ERROR: the query contains a repartition join, but repartition joins are currenly disabled
HINT: Set citus.enable_repartition_joins to on to enable repartition joins.
```

(we also want people to start forgetting about `citus.task_executor_type` as we prepare for a unified executor)

Reply:
or maybe:
```
ERROR: the query contains a join that requires repartitioning
HINT: Set citus.enable_repartition_joins to on to enable repartitioning
```

Reply:
done

Reply:
Cool, maybe squash into 1 commit and :ship: